webhook_to_fedora_messaging/endpoints/util.py

```diff
@@ -1,31 +1,44 @@
-from functools import wraps
-
-from flask import abort, request
-
-
-def validate_request(fields=None):
-    fields = fields if fields is not None else ["username"]
-
-    def decorator(func):
-        @wraps(func)
-        def wrapper(*args, **kwargs):
-            # Check if the request has JSON data
-            if not request.is_json:
-                return abort(415, {"error": "Invalid input, JSON required"})
-            data = request.get_json()
-            # Check if all required fields are present
-            missing_fields = [field for field in fields if field not in data]
-            if missing_fields:
-                return abort(400, {"error": f"Missing fields: {', '.join(missing_fields)}"})
-
-            return func(*args, **kwargs)
-
-        return wrapper
-
-    # If the decorator is used without arguments
-    if callable(fields):
-        func = fields
-        fields = ["username"]
-        return decorator(func)
-
-    return decorator
+from fastapi import Depends, HTTPException
+from sqlalchemy import select
+from sqlalchemy.exc import NoResultFound
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import selectinload
+from starlette.status import HTTP_403_FORBIDDEN, HTTP_404_NOT_FOUND, HTTP_422_UNPROCESSABLE_ENTITY
+
+from webhook_to_fedora_messaging.auth import user_factory
+from webhook_to_fedora_messaging.database import get_session
+from webhook_to_fedora_messaging.models.service import Service
+from webhook_to_fedora_messaging.models.user import User
+
+
+async def is_uuid_vacant(uuid: str) -> str:
+    if uuid.strip() == "":
+        raise HTTPException(HTTP_422_UNPROCESSABLE_ENTITY, "No service UUID provided")
+    return uuid.strip()
+
+
+async def return_service_from_uuid(
+    uuid: str,
+    session: AsyncSession = Depends(get_session)  # noqa : B008
+) -> Service:
+    query = select(Service).filter_by(uuid=uuid).options(selectinload("*"))
+    result = await session.execute(query)
+    try:
+        service = result.scalar_one()
+    except NoResultFound as expt:
+        raise HTTPException(
+            HTTP_404_NOT_FOUND,
+            f"Service with the requested UUID '{uuid}' was not found"
+        ) from expt
+    return service
+
+
+async def check_service_permission(
```

Comment:
Are there places where you'd want the service without checking for the user permissions? If not, then you can merge those two functions. (I don't see a situation where you'd want the service without checking its permissions)

Reply:
When sending a message (i.e., the `message` POST endpoint), we do not necessarily want to authenticate the endpoint using the username/password; rather, we would use the access token about the said service. In those cases, we check if the service exists but not to whom the service belongs.